CONTRIBUTING.md

```diff
@@ -1,18 +1,139 @@
-# Contributing
+# Contributing to ICU4C Project
+
+`ICU4X` is an open source project and welcomes everyone to participate.
+
+In order to provide meaningful contributions, it is important to familiarize yourself with a set of documents which describe the [structure](./docs/charter.md) and [style](./docs/style-guide.md) used by the project.
+
+The list of [open issues](triaging.md) represent the current focus points of the project, and the [backlog](https://github.com/unicode-org/icu4x/issues?q=is%3Aissue+is%3Aclosed+label%3Abacklog+sort%3Aupdated-desc) list issues which we don't have resources to work on right now, but would consider accepting if someone volunteered to work on them.
+
+Issues are open to everyone to discuss and can be used to jump-start Pull Requests intended for the project.
+
+In most cases, the first step is to find or file a new issue related to your planned contribution, discuss it, and once you received a feedback indicating that the pull request would be welcomed you can start working on it.
```

Comment:
The README file also includes an introduction for contributors. The two are similar but different.

Reply:
I don't see much of an overlap. README lists how to join the core development team (which I don't think should be there), but except of that, we can just link to `CONTRIBUTE.md` once we land this.

I'd prefer to not extend the scope and deal with fine-tuning of README.md in a separate issue.

Reply:
The README was fixed in #146.

Reply:
Ah I see. What do you suggest? Would you want me to change README here, or can we land this and work on README separately?

Reply:
I think see should keep the executive summary for contributing in the README, and link to this doc for more details. I have a slight preference to do that in the same PR, but you can land it without those changes if you prefer.

Reply:
Leave this as-is and follow up with another PR after you merge this one. #149